starlite/openapi/parameters.py

```diff
@@ -115,6 +116,71 @@ def create_parameter(
     )
 
 
+def get_recursive_handler_parameters(
+    field_name: str,
+    model_field: "ModelField",
+    dependencies: Dict[str, "Provide"],
+    route_handler: "BaseRouteHandler",
+    path_parameters: List[Dict[str, Any]],
+    generate_examples: bool,
+) -> List[Parameter]:
+    """Create and return parameters for a handler. If the provided field is not a dependency,
+    a normal parameter is created and returned as a list, otherwise
+    `create_parameter_for_handler()` is called to generate parameters for the dependency."""
+
+    if field_name not in dependencies:
+        return [
+            create_parameter(
+                model_field=model_field,
+                parameter_name=field_name,
+                path_parameters=path_parameters,
+                generate_examples=generate_examples,
+            )
+        ]
+    dependency_fields = cast(BaseModel, dependencies[field_name].signature_model).__fields__
+    return create_parameter_for_handler(route_handler, dependency_fields, path_parameters, generate_examples)
+
+
+def get_layered_parameter(
+    field_name: str,
+    signature_model_field: "ModelField",
+    layered_parameters: Dict[str, "ModelField"],
+    path_parameters: List[Dict[str, Any]],
+    generate_examples: bool,
+) -> Parameter:
+    """Create a layered parameter for a given signature model field.
+    Layer info is extracted from the provided `layered_parameters` dict
+    and set as the field's `field_info` attribute."""
+    layer_field_info = layered_parameters[field_name].field_info
+    signature_field_info = signature_model_field.field_info
+
+    field_info = layer_field_info
+    # allow users to manually override Parameter definition using Parameter
+    if signature_field_info.extra.get(EXTRA_KEY_IS_PARAMETER):
+        field_info = signature_field_info
+
+    field_info.default = (
+        signature_field_info.default
+        if signature_field_info.default not in [Undefined, Ellipsis]
```

Comment:
Please move `[Undefined, Ellipsis]` into a variable in global scope, and change from list to set to increase speed

Reply:
sure but i won't be able to resolve this until next month. if you guys want this sooner i won't mind to have it resolved by someone else